source/Sashimi.Tests/AzureWebAppServiceMessageHandlerFixture.cs

```diff
@@ -180,14 +182,64 @@ public void BuildEndpoint_WhenWebAppSlotNameAttributeIsNotMissing_ShouldReturnEn
             });
         }
 
+        [Test]
+        public void BuildEndpoint_WhenWorkerPoolIdAttributeIsProvided_ShouldLookupWorkerPool()
+        {
+            var messageProperties = GetMessageProperties();
+            messageProperties.Add(AzureWebAppServiceMessageNames.WorkerPoolIdOrNameAttribute, "Worker Pool 1");
+
+            var workerPoolId = "WorkerPools-1";
+            const string accountId = "Accounts-12";
+            var endpoint = serviceMessageHandler.BuildEndpoint(messageProperties,
+                                                               GetVariableDictionary(),
+                                                               _ => accountId,
+                                                               null,
+                                                               _ => workerPoolId,
+                                                               null);
+
+            AssertAzureWebAppEndpoint(endpoint, new ExpectedEndpointValues
+            {
+                AccountId = accountId,
+                WebAppName = messageProperties[AzureWebAppServiceMessageNames.WebAppNameAttribute],
+                ResourceGroupName = messageProperties[AzureWebAppServiceMessageNames.ResourceGroupNameAttribute],
+                WebAppSlotName = messageProperties[AzureWebAppServiceMessageNames.WebAppSlotNameAttribute],
+                WorkerPoolId = workerPoolId
+            });
+        }
+
+        [Test]
+        public void BuildEndpoint_WhenNoWorkerPoolIdAttributeIsProvided_ShouldUseWorkerPoolFromStep()
+        {
+            var messageProperties = GetMessageProperties();
+            messageProperties.Remove(AzureWebAppServiceMessageNames.WorkerPoolIdOrNameAttribute);
+            const string accountId = "Accounts-12";
+            var variableDictionary = GetVariableDictionary();
+            var endpoint = serviceMessageHandler.BuildEndpoint(messageProperties,
+                                                               variableDictionary,
+                                                               _ => accountId,
+                                                               null,
+                                                               _ => variableDictionary.Get(KnownVariables.WorkerPool.Id),
+                                                               null);
+
+            AssertAzureWebAppEndpoint(endpoint, new ExpectedEndpointValues
+            {
+                AccountId = accountId,
+                WebAppName = messageProperties[AzureWebAppServiceMessageNames.WebAppNameAttribute],
+                ResourceGroupName = messageProperties[AzureWebAppServiceMessageNames.ResourceGroupNameAttribute],
+                WebAppSlotName = messageProperties[AzureWebAppServiceMessageNames.WebAppSlotNameAttribute],
+                WorkerPoolId = variableDictionary.Get(KnownVariables.WorkerPool.Id)
```

Comment:
WorkerPoolId should be nullable. It may not be entirely visible in this change, but the VariableDictionary's Get method can return null, I'm actually working on a couple of PRs atm to make sure more of these nullable things are set up correctly.